modules/Bio/EnsEMBL/BaseAlignFeature.pm

```diff
@@ -1009,7 +1012,259 @@ sub _query_unit {
+          my $cur_type = $self->_get_mdz_chunk_type( $char_arr[$i] );
+          if($cur_type ne $prev_type) {
+            # We've found a new character class, push the
+            # current chunk on to the list
+            push @chunks, $current_chunk;
+            # Restart the current pile
+            $current_chunk = $char_arr[$i];
+           } else {
+             # Same character class, put it on the current pile
+             $current_chunk .= $char_arr[$i];
+           }
+           # Shift current type in to the prior slot
+           $prev_type = $cur_type;
+          }
+
+        # Post loop cleanup of the last piece
+        # This also handles an array of size one where
+        # we never entered the loop, that one element
+        # now gets pushed on the pile.
+        push @chunks, $current_chunk;
+      }
+    }
+  return \@chunks;
+}
+
+=head2 _get_mdz_alignment_length
+
+  Arg [1]    : array of strings
+  Description: calculate the alignment length from the given chunks
+  Returntype : array of strings
+  Exceptions : none
+  Caller     : internal
+  Status     : Stable
+
+=cut
+
+sub _get_mdz_alignment_length{
+  my $self = shift;
+
+  my $chunks = shift;
+  my $length = 0;
+   for(my $i=0; $i< scalar(@$chunks); $i++){
+     my $chunk = $$chunks[$i];
+     my $type = $self->_get_mdz_chunk_type($chunk);
+
+    if($type eq 'num'){
+      $length = $length + $chunk;
+    }elsif($type eq 'alpha'){
+      $length = $length + length($chunk);
+    }elsif($type eq 'del'){
+      #skip next chunk
+      $i++;
+    }
+
+  }
+  return $length;
+}
+
+=head2 _get_mdz_chunk_type
+
+  Arg [1]    : char
+  Description: get the chunk type
+  Returntype : string
+  Exceptions : none
+  Caller     : internal
+  Status     : Stable
+
+=cut
+
+sub _get_mdz_chunk_type{
+  my $self = shift;
+  my $char = shift;
+
+  my $type;
+  if($char eq '^'){
+    $type = "del";
+  } elsif($char =~ /\d+/){
+    $type = 'num';
+  } elsif($char =~ /\w+/){
+    $type = 'alpha';
+  }
+  return $type;
+}
+
+
+=head2 _mdz_alignment_string
+
+  Arg [1]    : input sequence
+  Arg [2]    : MD Z String for mismatching positions. Regex : [0-9]+(([A-Z]|\^[A-Z]+)[0-9]+)* (Refer:  SAM/BAM specification)
+               eg: MD:Z:96^RHKTDSFVGLMGKRALNS0V14
+  Example    : $pf->alignment_strings
+  Description: Allows to rebuild the alignment string of both the seq and hseq sequence
+  Returntype : array reference containing 2 strings
+               the first corresponds to seq
+               the second corresponds to hseq
+  Exceptions : none
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+# Try to get the sequence from translation object
+# mdz_string from current object
+sub _mdz_alignment_string {
+  my ($self, $input_seq, $mdz_string) = @_;
+
+  my $chunks = $self->_get_mdz_chunks($mdz_string);
+
+  my $target_seq = "";
+  my $query_seq = "";
+
+   #Handle first chunk, which is always a num
+   my $first_chunk = $$chunks[0];
```

Comment:
`$$chunks[0]`
is equivalent to the more modern and pleasant to the eye
`$chunks->[0]`

`$$` gets a bit confusing easily in my opinion, however it is a matter of style and both ways work... 
It's up to you what you want to use, but thought it would be useful to let you know both ways.